common/element_handle_options.go

```diff
@@ -495,3 +495,33 @@ func (o *ElementHandleWaitForElementStateOptions) Parse(ctx context.Context, opt
 	}
 	return nil
 }
+
+// ElementHandleDispatchEventOptions are options for ElementHandle.dispatchEvent.
+type ElementHandleDispatchEventOptions struct {
+	ElementHandleBasePointerOptions
+	Timeout time.Duration `json:"timeout"`
+}
+
+// NewElementHandleDispatchEventOptions returns a new ElementHandleDispatchEventOptions.
+func NewElementHandleDispatchEventOptions(defaultTimeout time.Duration) *ElementHandleDispatchEventOptions {
+	return &ElementHandleDispatchEventOptions{
+		Timeout: defaultTimeout,
+	}
+}
+
+// Parse the ElementHandleDispatchEvent options from a goja value.
+func (o *ElementHandleDispatchEventOptions) Parse(ctx context.Context, opts goja.Value) error {
+	if !isGojaValueExists(opts) {
+		return nil
+	}
+
+	gopts := opts.ToObject(k6ext.Runtime(ctx))
+	for _, k := range gopts.Keys() {
+		if k == "timeout" {
+			o.Timeout = time.Duration(gopts.Get(k).ToInteger()) * time.Millisecond
+			break
```

Comment:
`timeout` is already parsed in `ElementHandleBaseOptions`:
https://github.com/grafana/xk6-browser/blob/f97ca063a8a7f483faed0d2d01251d9d370c9af8/common/element_handle_options.go#L149-L150

So we could just return the error from `o.ElementHandleBasePointerOptions.Parse(ctx, opts)` here.

Though like I mentioned in a previous PR, that method doesn't actually return an error. So we should validate the received options and return an error on failure.

Reply:
Thanks, nice! I tried this and it worked. I'll be sending a PR shortly.

Regarding the validation, see #408.